src/nccl_ofi_sendrecv.c

```diff
@@ -2158,13 +2192,21 @@ static int device_prepare_for_connection(nccl_net_ofi_sendrecv_device_t *device)
 		goto error;
 	}
 
-	/* Create domain */
-	ret = fi_domain(device->fabric, device->info,
-			&device->domain, NULL);
-	if (OFI_UNLIKELY(ret != 0)) {
-		NCCL_OFI_WARN("Couldn't open a fabric access domain. RC: %d, ERROR: %s",
-			      ret, fi_strerror(-ret));
-		goto error;
+	/*
+	 * In the domain-per-device case, create the domain in the endpoint structure.  In the
+	 * domain-per-endpoint case, keep it in the device structure.  This is because, on some
+	 * platforms we lock when accessing the domain, so retaining separate domains per thread
+	 * and per endpoint reduces contention for that lock.
```

Comment:
> This is because, on some platforms we lock when accessing the domain

This comment confused me a bit. IIUC we (as in we the plugin) don't lock when accessing the domain, rather, Libfabric takes a lock when accessing the domain.

Reply:
Thanks; I'll clean it up.